server/base/src/main/java/org/apache/accumulo/server/metrics/MetricsSystemHelper.java

```diff
@@ -16,20 +16,65 @@
  */
 package org.apache.accumulo.server.metrics;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import org.apache.hadoop.metrics2.MetricsSystem;
 import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
+import org.apache.hadoop.metrics2.source.JvmMetrics;
+import org.apache.hadoop.metrics2.source.JvmMetricsInfo;
 
 /**
  *
  */
 public class MetricsSystemHelper {
 
+  private static Map<String,String> serviceNameMap = new HashMap<>();
+  private static String processName = "Unknown";
+
+  static {
+    serviceNameMap.put("master", "Master");
+    serviceNameMap.put("tserver", "TabletServer");
+    serviceNameMap.put("monitor", "Monitor");
+    serviceNameMap.put("gc", "GarbageCollector");
+    serviceNameMap.put("tracer", "Tracer");
+    serviceNameMap.put("shell", "Shell");
+  }
+
+  public static void configure(String application) {
+    String serviceName = application;
+    if (MetricsSystemHelper.serviceNameMap.containsKey(application)) {
+      serviceName = MetricsSystemHelper.serviceNameMap.get(application);
+    }
+
+    // a system property containing 'instance' can be used if more than one TabletServer is started on a host
+    String serviceInstance = "";
+    if (serviceName.equals("TabletServer")) {
+      for (Map.Entry<Object,Object> p : System.getProperties().entrySet()) {
+        if (((String) p.getKey()).contains("instance")) {
```

Comment:
This was an attempt to not tie the code to a specifc property.  It's probably better to just use the property.   I can't just pass all of the properties to the metrics system and defer the decision.  I'm creating a specific tag "ProcessName" that is used by the MetricsRegistry and metrics2 sinks.  We are using the -Daccumulo.service.instance to deconflict the log file names (this is why the _ is added to the instance) and to correlate a running process to the logs.  I'm trying to make it easy to montor the long-running tserver processes by using well-named metrics that also correlate to the running process and the log file with the same instance (1, 2, 3, etc).  Since this is only a concern for the tservers, I could grab this property closer to the top of the call chain (TabletServer.main()) if that would be more acceptable.